File: python/leetcode/382.py
class Solution(object):
    def __init__(self, head):
        self.head = head

    def getRandom(self):
        import random
        res, n, head = -1, 0, self.head
        while head:
            if random.randint(0, n) == 0:
                res = head.val
            head = head.next
            n += 1
        return res

File: cpp/leetcode/382.cpp
class Solution {
public:
    Solution(ListNode *head) {
        nhead = head;
    }

    int getRandom() {
        int cnts = 0, res = 0;
        ListNode *ptr = nhead;
        while (ptr) {
            cnts++;
            int j = rand() % cnts;
            if (j == 0) {
                res = ptr->val;
            }
            ptr = ptr->next;
        }
        return res;
    }

  private:
    ListNode *nhead;
};